src/Doctrine/ORM/PickupPointRepository.php

```diff
@@ -27,6 +23,7 @@ public function findOneByCode(PickupPointCode $code): ?PickupPointInterface
                 'codeProvider' => $code->getProviderPart(),
                 'codeCountry' => $code->getCountryPart(),
             ])
+            ->setMaxResults(1)
```

Comment:
Why this? Is this needed when we call `getOneOrNullResult` below?

Reply:
`getOneOrNullResult` may throw NonUniqueException otherwise. Or we should have some unique indexes on embedded fields

Reply:
Somehow I feel this is wrong. I would rather have the unique indexes then. Wdyt?

Reply:
Yeah, indexes is the best solution, but I haven't found how to add indexes to embeddable at doctrine docs

Reply:
My guess is you can do it listening to the Doctrine loadClassMetaData event?

Reply:
Will try that. Anyway, it looks like require to dive deep